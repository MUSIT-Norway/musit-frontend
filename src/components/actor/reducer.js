import { Observable } from 'rxjs/Rx';
import actions from "./actions";

const dataFromServer = [
  {
    id: 1,
    fn: 'Jarl parl'
  },
  {
    id: 2,
    fn: 'Line pine'
  },
  {
    id: 3,
    fn: 'Kjell pell'
  },
  {
    id: 4,
    fn: 'Stein sein'
  },
  {
    id: 5,
    fn: 'Knut lut'
  },
  {
    id: 6,
    fn: 'Ellen'
  }
]

const initialState = [];

const ActorReducer$ = Observable.of(() => initialState)
  .merge(
    actions.suggest$.map((update) => () => {
      if (!update) {
        return initialState;
      }
      return dataFromServer.filter(d => {
        const name = d.fn.toLowerCase();
        const query = update.value.toLowerCase();
        return name.indexOf(query) > -1
      })
    }),
    actions.clear$.map(() => () => initialState),
  );

export default Observable.merge(
  ActorReducer$.map(reducer => ["suggest", reducer]),
);