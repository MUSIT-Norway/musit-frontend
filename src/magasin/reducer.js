import { toFrontend } from './mapper';
import * as types from './types';

const initialState = {
  root: {
    data: {}
  }
};

const storageUnitGridReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.LOAD_SEVERAL:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SEVERAL_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.result
    };
  case types.LOAD_SEVERAL_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.LOAD_ONE_SUCCESS:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: true,
        data: toFrontend(action.result)
      }
    };
  case types.LOAD_ONE_FAIL:
    return {
      ...state,
      root: {
        ...state.root,
        loading: false,
        loaded: false,
        error: action.error
      }
    };
  case types.CLEAR_ROOT: {
    return {
      ...state,
      root: {}
    };
  }
  case types.DELETE:
    return {
      ...state,
      loading: true
    };
  case types.DELETE_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: []
    };
  case types.DELETE_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  default:
    return state;
  }
};

export default storageUnitGridReducer;