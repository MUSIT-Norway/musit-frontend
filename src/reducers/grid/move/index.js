const LOAD = 'musit/movehistory/LOAD'
const LOAD_SUCCESS = 'musit/movehistory/LOAD_SUCCESS'
const LOAD_FAIL = 'musit/movehistory/LOAD_FAIL'
const CLEAR = 'musit/movehistory/CLEAR_SUCCESS'
const CLEAR_SUCCESS = 'musit/movehistory/CLEAR'
const CLEAR_FAIL = 'musit/movehistory/CLEAR_FAIL'

const initialState = { data: [
  { doneBy: 'Kjell Kjellstad',
    doneDate: '23.12.2001',
    name: 'Arne And',
    from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
    to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
  },
  { doneBy: 'Bjørn Bjørnsen',
    doneDate: '21.12.2005',
    name: 'Arne And',
    from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
    to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
  },
  { doneBy: 'Jan Pettersen',
    doneDate: '12.01.2015',
    name: 'Arne And',
    from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
    to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
  },
  { doneBy: 'Kjell Kjellstad',
      doneDate: '23.12.2001',
      name: 'Arne And',
      from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
      to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
    },
    { doneBy: 'Bjørn Bjørnsen',
      doneDate: '21.12.2005',
      name: 'Arne And',
      from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
    },
    { doneBy: 'Jan Pettersen',
      doneDate: '12.01.2015',
      name: 'Arne And',
      from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
    },
    { doneBy: 'Kjell Kjellstad',
      doneDate: '23.12.2001',
      name: 'Arne And',
      from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Es55555555555555555555555555ke 3' },
      to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
    },
    { doneBy: 'Bjørn Bjørnsen',
      doneDate: '21.12.2005',
      name: 'Arne And',
      from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
    },
    { doneBy: 'Jan Pettersen',
      doneDate: '12.01.2015',
      name: 'Arne And',
      from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 355555555555555555555555555555555555555555555555555555555' }
    },
    { doneBy: 'Kjell Kjellstad',
      doneDate: '23.12.2001',
      name: 'Arne And',
      from: { path: 'NHM/3etg/Rom 3/Skap 4/Skuff 4/Eske 3' },
      to: { path: 'NHM/3etg/Rom 3/Skap 2/Skuff 1/Eske 8' }
    },
    { doneBy: 'Bjørn Bjørnsen',
      doneDate: '21.12.2005',
      name: 'Arne And',
      from: { path: 'NHM/1 etg/Rom 4/Skap 14/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Rom 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
    },
    { doneBy: 'Jan Pettersen',
      doneDate: '12.01.2015',
      name: 'Arne And',
      from: { path: 'KHM/1 etg/Rom 4/Skap 4/Skuff 1/Eske 10' },
      to: { path: 'NHM/3 etg/Ro5555555555555555555555555555555555555555555555555555555m 1/Skap 6/Skuff 4/Eske 18/Boks 3/Mappe 3' }
    }
] }

const moveHistoryReducer = (state = initialState, action) => {
  switch (action.type) {
    case LOAD:
      return {
        ...state,
        loading: true,
        loaded: false
      };
    case LOAD_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: true,
        data: action.result
      };
    case LOAD_FAIL:
      return {
        ...state,
        loading: false,
        loaded: false,
        error: action.error
      }
    case CLEAR_SUCCESS:
      return {
        ...state,
        loading: false,
        loaded: false,
        data: action.result
      }
    case CLEAR:
      return {
        ...state,
        loading: true,
        loaded: false
      }
    case CLEAR_FAIL:
      return {
        ...state,
        loading: false,
        loaded: false,
        error: action.error
      }
    default:
      return state;
  }
}

export default moveHistoryReducer;

export const loadMoveHistoryForObject = () => {
  return {
    type: LOAD_SUCCESS,
    result: initialState.data
  }
}
export const clearMoveHistoryForObject = () => {
  return {
    type: CLEAR_SUCCESS,
    result: []
  }
}
