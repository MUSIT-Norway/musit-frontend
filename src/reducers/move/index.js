
const MOVE_OBJECT = 'musit/move/object/start'
const MOVE_OBJECT_SUCCESS = 'musit/move/object/succes'
const MOVE_OBJECT_FAILURE = 'musit/move/object/failure'
const MOVE_NODE = 'musit/move/node/start'
const MOVE_NODE_SUCCESS = 'musit/move/node/succes'
const MOVE_NODE_FAILURE = 'musit/move/node/failure'

const ACTIONS = {
  [MOVE_OBJECT]: () => ({ loading: true, loaded: false }),
  [MOVE_OBJECT_SUCCESS]: () => ({ loading: false, loaded: true }),
  [MOVE_OBJECT_FAILURE]: (state, action) => ({ loading: false, loaded: false, error: action.error }),
  [MOVE_NODE]: () => ({ loading: true, loaded: false }),
  [MOVE_NODE_SUCCESS]: () => ({ loading: false, loaded: true }),
  [MOVE_NODE_FAILURE]: (state, action) => ({ loading: false, loaded: false, error: action.error })
}

export default (state = {}, action) => {
  if (ACTIONS[action.type]) {
    return ACTIONS[action.type](state, action)
  }
  return state
}

export const moveObject = (objectId, destination, doneBy, callback) => {
  const data = {
    doneBy,
    destination,
    items: [].concat(objectId)
  }
  return {
    types: [MOVE_OBJECT, MOVE_OBJECT_SUCCESS, MOVE_OBJECT_FAILURE],
    promise: (client) => client.put('/api/storagefacility/v1/storagenodes/moveObject', { data }),
    callback
  }
}

export const moveNode = (nodeId, destination, doneBy, callback) => {
  const data = {
    doneBy,
    destination,
    items: [].concat(nodeId)
  }
  return {
    types: [MOVE_OBJECT, MOVE_OBJECT_SUCCESS, MOVE_OBJECT_FAILURE],
    promise: (client) => client.put('/api/storagefacility/v1/storagenodes/moveNode', { data }),
    callback
  }
}
