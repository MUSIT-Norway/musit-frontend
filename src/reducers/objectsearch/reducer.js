import { omit } from 'lodash';
import { getPath } from '../helper';
import MusitObject from '../../models/object';
import * as types from './constants';

const initialState = {
  data: {
    totalMatches: 0,
    matches: []
  },
  params: {
    currentPage: 1,
    perPage: 50
  },
  loaded: true
};

export default (state = initialState, action) => {
  switch(action.type) {
  case types.CHANGE_FIELD:
    return {
      ...state,
      params: {
        ...state.params,
        [action.field]: action.value
      }
    };
  case types.SEARCH_OBJECTS:
    return {
      ...omit(state, 'error'),
      loading: true,
      loaded: false,
      data: {
        totalMatches: 0,
        matches: []
      },
      params: {
        ...state.params,
        currentPage: action.page
      }
    };
  case types.SEARCH_OBJECTS_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      data: [],
      error: action.error
    };
  case types.SEARCH_OBJECTS_SUCCESS:
    return {
      ...omit(state, 'error'),
      loading: false,
      loaded: true,
      data: {
        totalMatches: action.result.totalMatches,
        matches: action.result.matches.map(data => {
          return new MusitObject({
            ...data,
            breadcrumb: getPath(data)
          });
        })
      }
    };
  default:
    return state;
  }
};