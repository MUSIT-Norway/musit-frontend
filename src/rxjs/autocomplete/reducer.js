import { Observable } from 'rxjs/Rx';

export default function createReducer(actions) {
  const initialState = [];
  return Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
}