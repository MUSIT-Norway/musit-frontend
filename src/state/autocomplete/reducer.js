import { Observable } from 'rxjs/Rx';

export default function createReducer(actions, stateKey = "suggest") {
  const initialState = [];
  const reducer$ = Observable.of(() => initialState)
    .merge(
      actions.update$.map((data) => () => data),
      actions.clear$.map(() => () => initialState),
    );
  return Observable.merge(
    reducer$.map(reducer => [stateKey, reducer]),
  );
}