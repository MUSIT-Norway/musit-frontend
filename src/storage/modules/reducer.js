import { toFrontend } from '../../magasin/modules/mapper';
import * as types from './types';

const initialState = {
  state: {
    environmentRequirement: {},
    environmentAssessment: {},
    securityAssessment: {}
  }
};

const storageUnitContainerReducer = (state = initialState, action = {}) => {
  switch (action.type) {
  case types.INSERT:
    return {
      ...state,
      loading: true
    };
  case types.INSERT_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.INSERT_FAIL:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.error
    };
  case types.UPDATE:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: action.data
    };
  case types.LOAD:
    return {
      ...state,
      loading: true
    };
  case types.LOAD_SUCCESS:
    return {
      ...state,
      loading: false,
      loaded: true,
      data: toFrontend(action.result)
    };
  case types.LOAD_FAIL:
    return {
      ...state,
      loading: false,
      loaded: false,
      error: action.error
    };
  case types.UPDATE_STATE:
    return {
      ...state,
      state: action.data
    };
  case types.CLEAR_STATE:
    return {
      ...state,
      state: {
        environmentRequirement: {},
        environmentAssessment: {},
        securityAssessment: {}
      }
    };
  default:
    return state;
  }
};

export default storageUnitContainerReducer;