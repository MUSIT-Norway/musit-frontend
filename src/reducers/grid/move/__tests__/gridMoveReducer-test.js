
import * as actions from '../../../../reducers/grid/move/index'
import reducer from '../../../../reducers/grid/move/index'
import Config from '../../../../config'
import request from 'superagent';
import nocker from 'superagent-nock';
const nock = nocker(request);

describe('GridMoveHistory', () => {
    it('creates LOAD_ACTOR_SUCCESS when fetching data has been done', () => {
        const actorPostData = [1,2]
        const url = '/api/actor/v1/person/details'
        nock('http://localhost')
            .post(url, actorPostData)
            .reply(201, [
                {
                    id: 1,
                    fn: 'Jarle Stabell',
                    dataportenId: 'jarle'
                },
                {
                    id: 2,
                    fn: 'Stein A. Olsen',
                    dataportenId: 'stein'
                }
            ])
        const store = mockStore()

        return store.dispatch(actions.loadActor([1, 2]))
            .then(() => {
                expect(store.getActions()).toMatchSnapshot()
            })
    })

    it('no action', () => {
        expect(
            reducer(undefined, {})
        ).toMatchSnapshot()
    })

    it('initial action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD_ACTOR
            })
        ).toMatchSnapshot()
    })

    it('success action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD_ACTOR_SUCCESS,
                result: {
                    someField: 1
                }
            })
        ).toMatchSnapshot()
    })

    it('fail action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD_ACTOR_FAIL,
                error: Error('Some error here.')
            })
        ).toMatchSnapshot()
    })


    it('creates LOAD_SUCCESS when fetching data has been done', () => {
        const OutputSuccess = { data:
            [
                {
                    doneBy: 2,
                    doneDate: '2016-10-06T00:00:00+00:00',
                    registeredBy: 'Darth Vader',
                    registeredDate: '2016-10-06T05:55:52+00:00',
                    from: {
                        path: ',1,2,3,6,7,',
                        pathNames: [
                            {
                                name: 'root-node',
                                nodeId: 1
                            },
                            {
                                name: 'Utviklingsmuseet',
                                nodeId: 2
                            },
                            {
                                name: 'Forskningens hus',
                                nodeId: 3
                            },
                            {
                                name: 'Forskningsværelset',
                                nodeId: 6
                            },
                            {
                                name: 'Foo',
                                nodeId: 7
                            }
                        ]
                    },
                    to: {
                        path: ',1,2,3,6,',
                        pathNames: [
                            {
                                name: 'root-node',
                                nodeId: 1
                            },
                            {
                                name: 'Utviklingsmuseet',
                                nodeId: 2
                            },
                            {
                                name: 'Forskningens hus',
                                nodeId: 3
                            },
                            {
                                name: 'Forskningsværelset',
                                nodeId: 6
                            }
                        ]
                    }
                }]
        }
        const id = 1
        const url = `${Config.magasin.urls.storagefacility.baseUrl(1)}/objects/${id}/locations`
        nock('http://localhost')
            .get(url)
            .reply(200, OutputSuccess)
        const store = mockStore()

        return store.dispatch(actions.loadMoveHistoryForObject(1))
            .then(() => {
                expect(store.getActions()).toMatchSnapshot()
            })
    })

    it('Move history: no action', () => {
        expect(
            reducer(undefined, {})
        ).toMatchSnapshot()
    })

    it('Move history: initial action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD
            })
        ).toMatchSnapshot()
    })

    it('Move history: success action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD_SUCCESS,
                result: {
                    someField: 1
                }
            })
        ).toMatchSnapshot()
    })

    it('Move history: fail action', () => {
        expect(
            reducer(undefined, {
                type: actions.LOAD_FAIL,
                error: Error('Some error to load history data.')
            })
        ).toMatchSnapshot()
    })
})