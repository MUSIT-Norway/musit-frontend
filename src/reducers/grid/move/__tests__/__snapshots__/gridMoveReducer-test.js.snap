exports[`GridMoveHistory CLEAR_SUCCESS action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": false
}
`;

exports[`GridMoveHistory Move history: fail action 1`] = `
Object {
  "data": [Error: Some error to load history data.],
  "loaded": true,
  "loading": false
}
`;

exports[`GridMoveHistory Move history: initial action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": true
}
`;

exports[`GridMoveHistory Move history: no action 1`] = `
Object {
  "data": Array []
}
`;

exports[`GridMoveHistory Move history: success action 1`] = `
Object {
  "data": Object {
    "someField": 1
  },
  "loaded": true,
  "loading": false
}
`;

exports[`GridMoveHistory creates LOAD_ACTOR_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/movehistory/LOAD_ACTOR"
  },
  Object {
    "result": Array [
      Object {
        "dataportenId": "jarle",
        "fn": "Jarle Stabell",
        "id": 1
      },
      Object {
        "dataportenId": "stein",
        "fn": "Stein A. Olsen",
        "id": 2
      }
    ],
    "type": "musit/movehistory/LOAD_ACTOR_SUCCESS"
  }
]
`;

exports[`GridMoveHistory creates LOAD_SUCCESS when fetching data has been done 1`] = `
Array [
  Object {
    "type": "musit/movehistory/LOAD"
  },
  Object {
    "result": Object {
      "data": Array [
        Object {
          "doneBy": 2,
          "doneDate": "2016-10-06T00:00:00+00:00",
          "from": Object {
            "path": ",1,2,3,6,7,",
            "pathNames": Array [
              Object {
                "name": "root-node",
                "nodeId": 1
              },
              Object {
                "name": "Utviklingsmuseet",
                "nodeId": 2
              },
              Object {
                "name": "Forskningens hus",
                "nodeId": 3
              },
              Object {
                "name": "Forskningsværelset",
                "nodeId": 6
              },
              Object {
                "name": "Foo",
                "nodeId": 7
              }
            ]
          },
          "registeredBy": "Darth Vader",
          "registeredDate": "2016-10-06T05:55:52+00:00",
          "to": Object {
            "path": ",1,2,3,6,",
            "pathNames": Array [
              Object {
                "name": "root-node",
                "nodeId": 1
              },
              Object {
                "name": "Utviklingsmuseet",
                "nodeId": 2
              },
              Object {
                "name": "Forskningens hus",
                "nodeId": 3
              },
              Object {
                "name": "Forskningsværelset",
                "nodeId": 6
              }
            ]
          }
        }
      ]
    },
    "type": "musit/movehistory/LOAD_SUCCESS"
  }
]
`;

exports[`GridMoveHistory fail action 1`] = `
Object {
  "data": Object {
    "actor": [Error: Some error here.]
  },
  "loaded": true,
  "loading": false
}
`;

exports[`GridMoveHistory initial action 1`] = `
Object {
  "data": Array [],
  "loaded": false,
  "loading": true
}
`;

exports[`GridMoveHistory no action 1`] = `
Object {
  "data": Array []
}
`;

exports[`GridMoveHistory success action 1`] = `
Object {
  "data": Array [],
  "loaded": true,
  "loading": false
}
`;
